tests/Testcontainers.Tests/Fixtures/Images/DockerImageFixture.cs

```diff
@@ -18,6 +18,9 @@ public DockerImageFixture()
       Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", "1.0.0")), "myregistry.azurecr.io/baz/foo/bar:1.0.0");
       Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", string.Empty)), "myregistry.azurecr.io/baz/foo/bar");
       Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", string.Empty)), "myregistry.azurecr.io/baz/foo/bar:latest");
+      Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", "1.0.0@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")), "myregistry.azurecr.io/baz/foo/bar:1.0.0@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+      Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", "@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")), "myregistry.azurecr.io/baz/foo/bar@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+      Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", "latest@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")), "myregistry.azurecr.io/baz/foo/bar:latest@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
```

Comment:
The expected output is not clear to me. Should it be as is or 
```suggestion
      Add(new DockerImageFixtureSerializable(new DockerImage("myregistry.azurecr.io/baz/foo", "bar", "@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")), "myregistry.azurecr.io/baz/foo/bar:latest@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
```

as for other cases when latest is removed?

Reply:
I think if the digest is provided, we use that instead of the tag (do not append `latest`). If we get both, we use both. Does that make sense?

Reply:
For [DockerImageFixture](https://github.com/testcontainers/testcontainers-dotnet/blob/4918417876ea8847334c2247e06b76b8183ee8d2/tests/Testcontainers.Tests/Fixtures/Images/DockerImageFixture.cs), I came up with this approach. We can then assert the full name like [this](https://github.com/testcontainers/testcontainers-dotnet/blob/4918417876ea8847334c2247e06b76b8183ee8d2/tests/Testcontainers.Tests/Unit/Images/TestcontainersImageTest.cs#L71). I left it uncommented because the full name does not take the digest into account yet.

Reply:
Should we merge #1256 first and then update this PR to resolve the `FullName` correct (including the digest)? WDYT?

Reply:
I am fine with that. Feel free to merge.